File: Models/Definition/DestinyItemSocketBlockDefinition.cs
namespace bulkybook.Models;

public class DestinyItemSocketBlockDefinition
{
    public string detail {get;set;}
    public List<DestinyItemSocketEntryDefinition> socketEntries {get;set;}
    public List<DestinyItemIntrinsicSocketEntryDefinition> intrinsicSockets {get;set;}
    public List<DestinyItemSocketCategoryDefinition> socketCategories {get;set;}

}

File: Scripts/Models/DestinyItemSocketBlockDefinition.ts
import { DestinyItemSocketEntryDefinition } from "./DestinyItemSocketEntryDefinition";
import { DestinyItemIntrinsicSocketEntryDefinition } from "./DestinyItemIntrinsicSocketEntryDefinition";
import { DestinyItemSocketCategoryDefinition } from "./DestinyItemSocketCategoryDefinition";
export interface DestinyItemSocketBlockDefinition
{
    detail: string;
    socketEntries: Array<DestinyItemSocketEntryDefinition>;
    intrinsicSockets: Array<DestinyItemIntrinsicSocketEntryDefinition>;
    socketCategories:  Array<DestinyItemSocketCategoryDefinition>;
}